### ISSUE TITLE
error in oxygen 16.1 / saxon 9.5

### ISSUE THREAD

**Issue by tofi86** [dita-semia/XsltGui#5]
Hi Patrick,

the following error occurs for `<gui:open-file>` in oxygen 16.1 which ships with saxon 9.5:

```
Engine name: Saxon-EE 9.5.1.7
Severity: warning
Description: Invalid configuration property extensionElement. Supplied value
    'org.DitaSemia.XsltGui.SaxonFactory', required value is the name of a class
    that implements 'com.saxonica.xsltextn.ExtensionElementFactory':
    Failed to load org.DitaSemia.XsltGui.SaxonFactory
Start location: 17:52

Engine name: Saxon-EE 9.5.1.7
Severity: fatal
Description: XTDE1450: Unknown extension instruction
Start location: 100:0
URL: http://www.w3.org/TR/xslt20/#err-XTDE1450
```

What's the problem here?

Any chance to get this to work also with oXygen 16.1?

@hschmull and me would love to include awesome GUI's to our framework, but we need to also support oxygen 16.1... :-/

**Comment by PStellmann** [comment]
Saxon's interface for extension functions has changes significantly from version 9.5 to 9.6.

I can't imagine that it's possible to have an extension library that will work with both versions. But it should be possible to have a different library for the same xslt extension instructions that works with Saxon 9.5 and oXygen 16.1 so you can use the same xslt code and cofiguration file.

I will check this out - possibly this weekend.

**Comment by PStellmann** [comment]
I just pushed a new branch oxygen-16.1 where I made the required changes to work with oXygen 16.1. I also had to change some of the xslt code since I used XSLT 3.0 features that are not available there. But beside that it should be possible to create a framework that will work with oxygen 16.1 as well as newer releases.

**Comment by PStellmann** [comment]
BTW: I asked in the oxygen forum if there is a way to provide both libraries in the same framework and let oxygen decide automatically which one to use:
https://www.oxygenxml.com/forum/topic13212.html

**Comment by tofi86** [comment]
Thanks Patrik, that works fine in 16.1!
However, it isn't compatible with 17/18 - but that's what we kinda expected... ;-)

I'm closing this issue as it's resolved with 71c118ea158c7f2a6c46c157bd181e9f2d5aa0ac. Thanks.

I will reply to your forum post in a couple of minutes...
